src/library/vehicle.py

```diff
@@ -43,6 +43,9 @@ def __init__(self):
         # Rover status to make sure drop is completed before rtl
         self.winch_status = 0
 
+        self.lock = threading.Lock()
```

Comment:
A couple of lines of comments above this line would be helpful to explain why we have the locks